app/controllers/application_controller.rb

```diff
@@ -1,2 +1,46 @@
 class ApplicationController < ActionController::Base
+  include ApplicationHelper
+  before_action :configure_permitted_parameters, if: :devise_controller?
+
+  def require_login
```

Comment:
Devise actually creates theese helper methods, so this is not needed.
A good reed for the helper methods are here. [https://github.com/heartcombo/devise#controller-filters-and-helpers](https://github.com/heartcombo/devise#controller-filters-and-helpers)
We can use `before_action :authenticate_user!` instead.